conda/cli/main_remove.py

```diff
@@ -165,23 +167,13 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
     check_non_admin()
 
     if args.all and prefix == context.default_prefix:
-        msg = "cannot remove current environment. deactivate and run conda remove again"
+        msg = "Cannot remove current environment. Deactivate and run conda remove again"
         raise CondaEnvironmentError(msg)
 
-    if args.all and path_is_clean(prefix):
-        # full environment removal was requested, but environment doesn't exist anyway
-
-        # .. but you know what? If you call `conda remove --all` you'd expect the dir
-        # not to exist afterwards, would you not? If not (fine, I can see the argument
-        # about deleting people's work in envs being a very bad thing indeed), but if
-        # being careful is the goal it would still be nice if after `conda remove --all`
-        # to be able to do `conda create` on the same environment name.
-        #
-        # try:
-        #     rm_rf(prefix, clean_empty_parents=True)
-        # except:
-        #     log.warning("Failed rm_rf() of partially existent env {}".format(prefix))
+    if args.all and not is_conda_environment(prefix):
```

Comment:
Don't we _always_ want to check if the prefix is valid? Shouldn't it still fail here if I say `conda remove -n missing zlib`?